pkg/controllers/federatedhpa/controller.go

```diff
@@ -0,0 +1,543 @@
+			}
+			return worker.StatusError
+		}
+	}
+
+	if isFedHPAObjectUpdated {
+		logger.V(1).Info("Updating fed hpa object")
+		if _, err = fedobjectadapters.Update(ctx, f.fedClient.CoreV1alpha1(), fedHPAObject, metav1.UpdateOptions{}); err != nil {
+			errMsg := "Failed to update fed hpa object"
+			logger.Error(err, errMsg)
+			f.eventRecorder.Eventf(fedHPAObject, corev1.EventTypeWarning, EventReasonUpdateHPAFedObject,
+				errMsg+" %v, err: %v, retry later", fedHPAObject, err)
+			if apierrors.IsConflict(err) {
+				return worker.StatusConflict
+			}
+			return worker.StatusError
+		}
+	}
+
+	return worker.StatusAllOK
+}
+
+func (f *FederatedHPAController) handleCache(ctx context.Context, key Resource) (context.Context, *handleCacheResp, worker.Result) {
+	hpaFTC, exists := f.informerManager.GetResourceFTC(key.gvk)
+	if !exists {
+		// Waiting for func enqueueFedHPAObjectsForFTC enqueue it again.
+		return ctx, nil, worker.StatusAllOK
+	}
+
+	hpaGVR := hpaFTC.GetSourceTypeGVR()
+	ctx, logger := logging.InjectLoggerValues(ctx, "hpa-ftc", hpaFTC.Name)
+
+	lister, hasSynced, exists := f.informerManager.GetResourceLister(key.gvk)
+	if !exists {
+		return ctx, nil, worker.StatusAllOK
+	}
+	if !hasSynced() {
+		// If lister is not yet synced, simply reenqueue after a short delay.
+		logger.V(3).Info("Lister for source hpa type not yet synced, will reenqueue")
+		return ctx, nil, worker.Result{
+			Success:      true,
+			RequeueAfter: pointer.Duration(f.cacheSyncRateLimiter.When(key)),
+		}
+	}
+	f.cacheSyncRateLimiter.Forget(key)
+
+	hpaUns, err := lister.Get(key.QualifiedName().String())
+	if err != nil {
+		logger.Error(err, "Failed to get source hpa object from store")
+		return ctx, nil, worker.StatusAllOK
+	}
+	hpaObject := hpaUns.(*unstructured.Unstructured).DeepCopy()
+
+	fedHPAObjectName := naming.GenerateFederatedObjectName(hpaObject.GetName(), hpaFTC.Name)
+	ctx, logger = logging.InjectLoggerValues(ctx, "federated-hpa-object", fedHPAObjectName)
+
+	fedHPAObject, err := fedobjectadapters.GetFromLister(
+		f.fedObjectInformer.Lister(),
+		nil,
+		hpaObject.GetNamespace(),
+		fedHPAObjectName,
+	)
+	if err != nil {
+		logger.Error(err, "Failed to get federated hpa object from store")
+		return ctx, nil, worker.StatusError
+	}
+
+	if ok, err := pendingcontrollers.ControllerDependenciesFulfilled(fedHPAObject, PrefixedFederatedHPAControllerName); err != nil {
+		logger.Error(err, "Failed to check controller dependencies")
+		return ctx, nil, worker.StatusError
+	} else if !ok {
+		return ctx, nil, worker.StatusAllOK
+	}
+
+	newWorkloadResource, err := f.scaleTargetRefToResource(key.gvk, hpaObject)
+	if err != nil {
+		logger.Error(err, "Failed to get workload resource from hpa")
+		return ctx, nil, worker.StatusError
+	}
+
+	ctx, logger = logging.InjectLoggerValues(ctx,
+		"workload-object", newWorkloadResource.QualifiedName(),
+		"workload-gvk", newWorkloadResource.gvk)
+
+	oldWorkloadResource, exist := f.workloadHPAMapping.LookupByT2(key)
```

Comment:
Is it possible here that oldWorkloadResource==newWorkloadResource?

Reply:
It is possible. I will make the update when oldWorkloadResource!=newWorkloadResource.

Reply:
The code has been updated.